src/main/java/seedu/momentum/model/project/Project.java

```diff
@@ -90,6 +91,23 @@ public Deadline getDeadline() {
         return deadline;
     }
 
+    /**
+     * Gets Deadline and name for {@code DeadLineCompare} comparator.
+     * Returns null if deadline is empty as {@code Comparator.nullsLast} method is used.
+     */
+    public HashMap<String, Object> getNullOrDeadline() {
+        if (deadline.isEmpty()) {
+            return null;
+        }
+        String name = getName().fullName;
+        Deadline deadline = getDeadline();
+        HashMap<String, Object> nameDeadlineMap = new HashMap<>();
+        nameDeadlineMap.put("name", name);
+        nameDeadlineMap.put("deadline", deadline);
+
+        return nameDeadlineMap;
```

Comment:
I'm not sure if there's a cleaner way to do this. What do the rest think?

Reply:
The reason why I need to use HashMap is because not all projects have Deadlines, and I'm using Comparator.nullsLast to place all projects without deadlines at the end of the list after projects with deadlines. But after extracting the deadlines (or nulls), I will still need to compare the names alphabetically in case of same deadline or null deadline, which is why I need to return the name as well. Anyone can lmk if there is a cleaner way to do this!